scene/3d/node_3d.cpp

```diff
@@ -88,7 +88,9 @@ void Node3D::_update_local_transform() const {
 	if (this->get_rotation_edit_mode() != ROTATION_EDIT_MODE_BASIS) {
 		data.local_transform = data.local_transform.orthogonalized();
 	}
-	data.local_transform.basis.set_euler_scale(data.rotation, data.scale);
+	if (this->get_rotation_edit_mode() == ROTATION_EDIT_MODE_EULER) {
+		data.local_transform.basis.set_euler_scale(data.rotation, data.scale);
```

Comment:
The whole code of this function looks wrong for some reason. Why setting local_transform if its going to be overriden later? Also setting the euler like this seems to ignore the rotation order of the node.

Reply:
Indeed. It seems that DIRTY_LOCAL is not needed in the first place. I will attempt to refactor it.